src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBody.java

```diff
@@ -0,0 +1,162 @@
+    private boolean hearingIsRequired;
+
+    private CCDDirectionPartyType docUploadForParty;
+
+    private CCDDirectionPartyType eyewitnessUploadForParty;
+
+    private String preferredCourtName;
+
+    private String preferredCourtAddress;
+
+    private CCDHearingDurationType estimatedHearingDuration;
+
+    private String hearingStatement;
+
+    private List<OrderDirection> otherDirectionList;
+
+    @Builder
+    @EqualsAndHashCode
+    private static class Party {
+        private String partyName;
+    }
+
+    @Builder
+    @EqualsAndHashCode
+    private static class Judicial {
+        private String firstName;
+
+        private String lastName;
+    }
+
+    @Builder
+    @EqualsAndHashCode
+    private static class OrderDirection {
+        @JsonProperty("OtherDirection")
+        private OtherDirection otherDirection;
+    }
+
+    @Builder
+    @EqualsAndHashCode
+    private static class OtherDirection {
+        private CCDOrderDirectionType extraOrderDirection;
+        private String directionComment;
+        private LocalDate sendBy;
+        private CCDDirectionPartyType forParty;
+        private final String hmctsURL = HMCTS_URL;
+    }
+
+    public static DocAssemblyTemplateBody from(CCDCase ccdCase, UserDetails userDetails) {
+        return DocAssemblyTemplateBody.builder()
+            .claimant(Party.builder()
+                .partyName(ccdCase.getApplicants()
+                    .get(0)
+                    .getValue()
+                    .getPartyName())
+                .build())
+            .defendant(Party.builder()
+                .partyName(ccdCase.getRespondents()
+                    .get(0)
+                    .getValue()
+                    .getPartyName())
+                .build())
+            .judicial(Judicial.builder()
+                .firstName(userDetails.getForename())
+                .lastName(userDetails.getSurname().orElse(""))
+                .build())
+            .currentDate(LocalDate.now())
+            .referenceNumber(ccdCase.getReferenceNumber())
+            .hasFirstOrderDirections(
+                ccdCase.getDirectionList().contains(CCDOrderDirectionType.DOCUMENTS))
+            .docUploadDeadline(
+                ccdCase.getDocUploadDeadline())
+            .hasSecondOrderDirections(
+                ccdCase.getDirectionList().contains(CCDOrderDirectionType.EYEWITNESS))
+            .eyewitnessUploadDeadline(
+                ccdCase.getEyewitnessUploadDeadline())
+            .docUploadForParty(
+                ccdCase.getDocUploadForParty())
+            .eyewitnessUploadForParty(
+                ccdCase.getEyewitnessUploadForParty())
+            .hearingIsRequired(
+                ccdCase.getHearingIsRequired().toBoolean())
+            .preferredCourtName(
+                "Some court")    // will be populated when the acceptance criterias are refined
+            .preferredCourtAddress(
+                "this is an address EC2Y 3ND")
```

Comment:
Mistake? or atleast make it as UNKOWN

Reply:
we have a separate story for the address, so I have just a placeholder there. It will be rendered in the template